crates/wasi-provider/src/wasi_runtime.rs

```diff
@@ -1,9 +1,17 @@
 use std::collections::HashMap;
-use std::fs::File;
+use std::io::SeekFrom;
 use std::path::Path;
 
 use failure::{bail, format_err};
-use log::info;
+use k8s_openapi::api::core::v1::{
```

Comment:
I'm not sure if we should be bringing the k8s types into providers, but it felt like it would work here. The other option I thought of was adding a simplified version of the status types into `kubelet` with a `From` implementation to turn them into the k8s types

Reply:
This seems trivial enough that adding this dependency to the crate doesn't seem worth it. Duplicating that definition seems like a better idea

Reply:
After seeing it in use, I definitely think we should not be using the k8 types here.

Reply:
I was gonna disagree with Ryan on this one, on the grounds that there's not case in which this isn't Kubernetes centric. But after reading the PR, I agree: It is probably better to just do a more concise status message inside a provider, and have an the Kubelet layer translate that into the clunky Kubernetes/OpenAPI format. Seems like a nice clean `enum` could do the trick.

That said, I'm definitely not set on this. If it turns out that importing Kube stuff makes the code less repetitive...

Reply:
This is all done